Add GPU resources for containers and output variables for exec

nicholasjackson opened the pull request:
## New Features:
Adds outputs for 'exec' resources

Exec resources now have a new parameter 'output' which is a map of key value pairs.
Values for output can be set by echoing a key value to the file '${EXEC_OUTPUT}' in 
the defined script for either remote or local exec.

```hcl
resource "exec" "install" {
  # Add the output
  echo "exec=install" >> $EXEC_OUTPUT
  echo "foo=bar" >> $EXEC_OUTPUT
  EOF

  timeout = "30s"
}

output "local_exec_install" {
  value = resource.exec.install.output.exec
}
```

Enable experimental support for nvidia GPUs for container resources

This feature configures the container to use the nvidia runtime and the nvidia
device plugin to access the GPU.  Currently this has only been tested with WSL2 and
Nvidia GPUs.

```hcl
resource "container" "gpu_test" {
  image {
    name = "nvcr.io/nvidia/k8s/cuda-sample:nbody"
  }

  command = ["nbody", "-gpu", "-benchmark"]

  resources {
    gpu {
      driver     = "nvidia"
      device_ids = ["0"]
    }
  }
}
```

eveld commented on pkg/config/resources/exec/provider.go:
```diff
@@ -61,23 +61,53 @@ func (p *Provider) Create(ctx context.Context) error {
 
 	p.log.Info("executing script", "ref", p.config.Meta.ID, "script", p.config.Script)
 
+	outPath := fmt.Sprintf("%s/%s.out", os.TempDir(), p.config.Meta.ID)
```
Just for future reference, should we write these temp files to os temp or jumppad temp?

nicholasjackson replied:
I saw we had jp temp after creating this, I might change it and push later.

eveld commented on pkg/config/resources/exec/resource.go:
```diff
@@ -34,9 +33,9 @@ type Exec struct {
-	Output   cty.Value `hcl:"output,optional"`                               // output values returned from Terraform
+	PID      int               `hcl:"pid,optional" json:"pid,omitempty"`             // PID stores the ID of the created connector service if it is a local exec
+	ExitCode int               `hcl:"exit_code,optional" json:"exit_code,omitempty"` // Exit code of the process
+	Output   map[string]string `hcl:"output,optional" json:"output,omitempty"`       // output values returned from Terraform
```
Should we update the comment to say Exec? :D

eveld reviewed (commented):
Just put some questions, nothing blocking, can merge and fix later if needed :)

nicholasjackson commented:
Cool ta, I have updated everything to use JumppadTemp rather than os.Temp as JumppadTemp is removed when down is called. At least it should have been, fixed that too.

eveld reviewed (approved):
LGTM
